sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,683 @@
+   * a value that represents the contents of a boot stage, e.g. a (truncated)
+   * hash, or a tag.
+   *
+   * If it is a hash, changes in a boot stage will change the secret value, and
+   * consequently the versioned keys and identity seeds generated at subsequent
+   * boot stages. If it is a tag, those secret values, versioned keys, and
+   * identity seeds will be preserved across updates of the boot stage as long
+   * as the tag remains the same.
+   *
+   * For transitioning to CreatorRootKey state, the length of this array must be
+   * `kDifKeymgrBindingValueLenRomExtHash`. For all other states, its length
+   * must be `kDifKeymgrBindingValueLenTag`.
+   */
+  const uint32_t *binding_value;
+
+  /**
+   * Maximum allowed version for keys generated at the next state.
+   */
+  uint32_t max_key_version;
+} dif_keymgr_state_params_t;
+
+/**
+ * Sets parameters for a key manager state.
+ *
+ * This function sets and locks the max version value and the binding value that
+ * is used to compute the secret value for the operational states of key
+ * manager, i.e. CreatorRootKey, OwnerIntermediateKey, and OwnerRootKey.
+ *
+ * @param keymgr A key manager handle.
+ * @param state A key manager state.
+ * @param params State parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_lockable_result_t dif_keymgr_set_state_params(
+    const dif_keymgr_t *keymgr, dif_keymgr_state_t state,
+    dif_keymgr_state_params_t params);
+
+/**
+ * Advances key manager state.
+ *
+ * This function initiates a transition to the next state, i.e. Initialized ->
+ * CreatorRootKey -> OwnerIntermediateKey -> OwnerRootKey -> Disabled. State
+ * transitions are irreversible during a power cycle. Key manager must be
+ * initialized using `dif_keymgr_initialize_state()` and parameters for the next
+ * state must be set using `dif_keymgr_set_state_params()`  prior to calling
+ * this function.
+ *
+ * This is an asynchronous function because key manager state transitions
+ * involve KMAC operations that can take some time to complete. Clients must
+ * check the status of key manager using `dif_keymgr_get_status()` before
+ * calling other functions in this library.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_advance_state(const dif_keymgr_t *keymgr);
+
+/**
+ * Disables key manager.
+ *
+ * This function disables key manager until the next power cycle by making it
+ * transition to Disabled state. Disabled state is a terminal state where key
+ * manager is no longer operational and its secret value is a random value.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_lockable_result_t dif_keymgr_disable(const dif_keymgr_t *keymgr);
+
+/**
+ * Status code bit flags.
+ *
+ * See also: `dif_keymgr_status_codes_t`.
+ */
+typedef enum dif_keymgr_status_code {
```

Comment:
Yes, you're right. It looks like I named the error values looking at the description in the spec instead of the register table. `kDifKeymgrStatusInvalidInput` should have been `kDifKeymgrStatusInvalidKmacCommand`.

Regarding translation: Yes, there is a one-to-one mapping between the bits of the `ERROR_CODE` register and the first four bits of `dif_keymgr_status_codes_t`. The fifth bit is defined as `OP_STATUS != WIP`. Other values of `OP_STATUS` is used to determine if we need to check the `ERROR_CODE` register or not.